src/trie.rs

```diff
@@ -237,7 +242,7 @@ where
                 )
             }
             Node::Branch(mut branch) => {
-                if partial.is_empty() {
+                if partial.at(0) == 16 {
```

Comment:
I do not understand the nibble completely. What's the meaning of the `partial.at(0) == 16`?

Reply:
If `partial.at(0) == 16` and the value is stored at the last location of the branch.

Reply:
In this implementation, branch has a 16-length node array and a simple value

Reply:
> I do not understand the nibble completely. What's the meaning of the `partial.at(0) == 16`?

16 is the terminate signal of a nibble, the `insert_at` method will remove the prefix when recursively call itself. When the partial is only left with 16, it means all the chars in the nibble are used along the path, so the value will be set as the branch value.

Reply:
> > I do not understand the nibble completely. What's the meaning of the `partial.at(0) == 16`?
> 
> 16 is the terminate signal of a nibble, the `insert_at` method will remove the prefix when recursively call itself. When the partial is only left with 16, it means all the chars in the nibble are used along the path, so the value will be set as the branch value.

I see.